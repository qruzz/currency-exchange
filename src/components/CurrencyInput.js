import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * This function returns a CurrencyInput component to be rendered. The component
 * is controlled by the arguments, handling only the rendering of the information.
 * @param	{string}	to				The currency to convert to
 * @param	{string}	from			The currency to convert from
 * @param	{string}	value			The value of the input field
 * @param	{func}		handleOnChange	The function to be fired on input change
 * @param	{string}	currency		The currency shorthand (eg. GBP)
 * @param	{func}		changeCurrency	The function to be fired changing the currency
 * @param	{object}	balance			The object containing the account balances
 * @param	{object}	style			The style object containing specific styles
 * @returns	{jsx}						Returns the CurrencyInput component
 */
function CurrencyInput({
	to,
	from,
	value,
	handleOnChange,
	currency,
	changeCurrency,
	balance,
	style,
}) {
	// Dynamic import of the icon based on which currency is used
	const icon = require(`../resources/icons/${currency}.svg`);

	// Disable the 'to' input
	const disabled = to === 'exchangeFromAmount';

	// Initiate the actual balance
	let actualBalance = 0;

	// If a balance is passed through, set the actual balance based on the currency in use
	if (balance) {
		actualBalance = balance[currency];
	}

	return (
		<Wrapper style={style}>
			<InnerWrapper>
				<Indicator>
					<IndicatorButton onClick={changeCurrency}>
						<Icon size={'large'} src={icon} />
						<p>{currency}</p>
					</IndicatorButton>
					<Balance>Balance: {actualBalance}</Balance>
				</Indicator>
				<Input
					type={'number'}
					placeholder={'0'}
					value={value}
					disabled={disabled}
					onChange={(event) => {
						handleOnChange(to, from, event);
					}}
				/>
			</InnerWrapper>
		</Wrapper>
	);
}

// Verify the passed through Props
CurrencyInput.propTypes = {
	value: PropTypes.string.isRequired,
	handleOnChange: PropTypes.func.isRequired,
	to: PropTypes.string.isRequired,
	from: PropTypes.string.isRequired,
	changeCurrency: PropTypes.func.isRequired,
	currency: PropTypes.string,
	balance: PropTypes.shape({
		GBP: PropTypes.string,
		EUR: PropTypes.string,
		USD: PropTypes.string,
	}),
	style: PropTypes.shape({
		background: PropTypes.string,
		color: PropTypes.string,
	}),
};

// Define the default PropTypes
CurrencyInput.defaultProps = {
	currency: 'GBP',
	balance: null,
	style: {
		background: '#FFFFFF',
		color: 'transparent',
	},
};

const Wrapper = styled.div`
	display: flex;
	flex-direction: column;
	width: 100%;
	height: 10vh;
	background: ${props => props.style.color};
	padding: 5% 0;

	@media (min-width: 700px) {
		width: 45%;
	}
`;

const InnerWrapper = styled.div`
	display: flex;
	align-items: center;
	height: 100%;
	padding: 0 4%;
`;

const Icon = styled.img`
	width: 100%;
	margin-right: 15px;
	height: ${props => (props.size === 'small' ? '10%' : '30%')}
`;

const Input = styled.input`
	width: 100%;
	font-size: 3rem; 
	appearance: none;
	margin: 0;
	background-color: transparent;
	text-align: right;
`;

const Indicator = styled.div`
	display: flex;
	flex-direction: column;
	height: 100%;
	width: 30px;
`;

const IndicatorButton = styled.button`
	display: flex;
	align-items: center;
	height: 100%;
	background-color: transparent;
	cursor: pointer;
`;

const Balance = styled.div`
	width: 20vw;
	color: black;
`;

export default CurrencyInput;
