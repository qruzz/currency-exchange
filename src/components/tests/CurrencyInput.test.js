import React from 'react';
import { shallow } from '../../enzyme';

import CurrencyInput from '../CurrencyInput';

/* eslint no-undef: 0 */

describe('CurrencyInput test', () => {
	it('should be defined', () => {
		expect(CurrencyInput).toBeDefined();
	});

	it('renders an input', () => {
		const wrapper = shallow(<CurrencyInput />);
		expect(wrapper.find(CurrencyInput)).toMatchSnapshot();
	});
});
