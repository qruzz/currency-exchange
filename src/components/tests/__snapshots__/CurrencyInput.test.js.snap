// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyInput test renders an input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <CurrencyInput
      balance={null}
      currency="GBP"
      style={
        Object {
          "background": "#FFFFFF",
          "color": "transparent",
        }
      }
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <ForwardRef>
            <ForwardRef>
              <ForwardRef
                size="large"
                src={
                  Object {
                    "ReactComponent": [Function],
                    "__esModule": true,
                    "default": "GBP.svg",
                  }
                }
              />
              <p>
                GBP
              </p>
            </ForwardRef>
            <ForwardRef>
              Balance: 
              0
            </ForwardRef>
          </ForwardRef>
          <ForwardRef
            disabled={false}
            onChange={[Function]}
            placeholder="0"
            type="number"
          />
        </ForwardRef>,
        "style": Object {
          "background": "#FFFFFF",
          "color": "transparent",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              <ForwardRef>
                <ForwardRef
                  size="large"
                  src={
                    Object {
                      "ReactComponent": [Function],
                      "__esModule": true,
                      "default": "GBP.svg",
                    }
                  }
                />
                <p>
                  GBP
                </p>
              </ForwardRef>
              <ForwardRef>
                Balance: 
                0
              </ForwardRef>
            </ForwardRef>,
            <ForwardRef
              disabled={false}
              onChange={[Function]}
              placeholder="0"
              type="number"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef>
                  <ForwardRef
                    size="large"
                    src={
                      Object {
                        "ReactComponent": [Function],
                        "__esModule": true,
                        "default": "GBP.svg",
                      }
                    }
                  />
                  <p>
                    GBP
                  </p>
                </ForwardRef>,
                <ForwardRef>
                  Balance: 
                  0
                </ForwardRef>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <ForwardRef
                      size="large"
                      src={
                        Object {
                          "ReactComponent": [Function],
                          "__esModule": true,
                          "default": "GBP.svg",
                        }
                      }
                    />,
                    <p>
                      GBP
                    </p>,
                  ],
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "size": "large",
                      "src": Object {
                        "ReactComponent": [Function],
                        "__esModule": true,
                        "default": "GBP.svg",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-htpNat",
                        "isStatic": false,
                        "rules": Array [
                          "
	width: 100%;
	margin-right: 15px;
	height: ",
                          [Function],
                          "
",
                        ],
                      },
                      "displayName": "styled.img",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-htpNat",
                      "target": "img",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    },
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "GBP",
                    },
                    "ref": null,
                    "rendered": "GBP",
                    "type": "p",
                  },
                ],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-EHOje",
                    "isStatic": true,
                    "rules": Array [
                      "
	display: flex;
	align-items: center;
	height: 100%;
	background-color: transparent;
	cursor: pointer;
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-EHOje",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "Balance: ",
                    0,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Balance: ",
                  0,
                ],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bZQynM",
                    "isStatic": true,
                    "rules": Array [
                      "
	width: 20vw;
	color: black;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bZQynM",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": true,
                "rules": Array [
                  "
	display: flex;
	flex-direction: column;
	height: 100%;
	width: 30px;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": false,
              "onChange": [Function],
              "placeholder": "0",
              "type": "number",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
	width: 100%;
	font-size: 3rem; 
	appearance: none;
	margin: 0;
	background-color: transparent;
	text-align: right;
",
                ],
              },
              "displayName": "styled.input",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "input",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
	display: flex;
	align-items: center;
	height: 100%;
	padding: 0 4%;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
	display: flex;
	flex-direction: column;
	width: 100%;
	height: 10vh;
	background: ",
            [Function],
            ";
	padding: 5% 0;

	@media (min-width: 700px) {
		width: 45%;
	}
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef>
            <ForwardRef>
              <ForwardRef>
                <ForwardRef
                  size="large"
                  src={
                    Object {
                      "ReactComponent": [Function],
                      "__esModule": true,
                      "default": "GBP.svg",
                    }
                  }
                />
                <p>
                  GBP
                </p>
              </ForwardRef>
              <ForwardRef>
                Balance: 
                0
              </ForwardRef>
            </ForwardRef>
            <ForwardRef
              disabled={false}
              onChange={[Function]}
              placeholder="0"
              type="number"
            />
          </ForwardRef>,
          "style": Object {
            "background": "#FFFFFF",
            "color": "transparent",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                <ForwardRef>
                  <ForwardRef
                    size="large"
                    src={
                      Object {
                        "ReactComponent": [Function],
                        "__esModule": true,
                        "default": "GBP.svg",
                      }
                    }
                  />
                  <p>
                    GBP
                  </p>
                </ForwardRef>
                <ForwardRef>
                  Balance: 
                  0
                </ForwardRef>
              </ForwardRef>,
              <ForwardRef
                disabled={false}
                onChange={[Function]}
                placeholder="0"
                type="number"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <ForwardRef>
                    <ForwardRef
                      size="large"
                      src={
                        Object {
                          "ReactComponent": [Function],
                          "__esModule": true,
                          "default": "GBP.svg",
                        }
                      }
                    />
                    <p>
                      GBP
                    </p>
                  </ForwardRef>,
                  <ForwardRef>
                    Balance: 
                    0
                  </ForwardRef>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <ForwardRef
                        size="large"
                        src={
                          Object {
                            "ReactComponent": [Function],
                            "__esModule": true,
                            "default": "GBP.svg",
                          }
                        }
                      />,
                      <p>
                        GBP
                      </p>,
                    ],
                    "onClick": undefined,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "size": "large",
                        "src": Object {
                          "ReactComponent": [Function],
                          "__esModule": true,
                          "default": "GBP.svg",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-htpNat",
                          "isStatic": false,
                          "rules": Array [
                            "
	width: 100%;
	margin-right: 15px;
	height: ",
                            [Function],
                            "
",
                          ],
                        },
                        "displayName": "styled.img",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-htpNat",
                        "target": "img",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      },
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "GBP",
                      },
                      "ref": null,
                      "rendered": "GBP",
                      "type": "p",
                    },
                  ],
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-EHOje",
                      "isStatic": true,
                      "rules": Array [
                        "
	display: flex;
	align-items: center;
	height: 100%;
	background-color: transparent;
	cursor: pointer;
",
                      ],
                    },
                    "displayName": "styled.button",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-EHOje",
                    "target": "button",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      "Balance: ",
                      0,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Balance: ",
                    0,
                  ],
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bZQynM",
                      "isStatic": true,
                      "rules": Array [
                        "
	width: 20vw;
	color: black;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bZQynM",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  },
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": true,
                  "rules": Array [
                    "
	display: flex;
	flex-direction: column;
	height: 100%;
	width: 30px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disabled": false,
                "onChange": [Function],
                "placeholder": "0",
                "type": "number",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "rules": Array [
                    "
	width: 100%;
	font-size: 3rem; 
	appearance: none;
	margin: 0;
	background-color: transparent;
	text-align: right;
",
                  ],
                },
                "displayName": "styled.input",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
	display: flex;
	align-items: center;
	height: 100%;
	padding: 0 4%;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "rules": Array [
              "
	display: flex;
	flex-direction: column;
	width: 100%;
	height: 10vh;
	background: ",
              [Function],
              ";
	padding: 5% 0;

	@media (min-width: 700px) {
		width: 45%;
	}
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <ForwardRef>
            <ForwardRef>
              <ForwardRef
                size="large"
                src={
                  Object {
                    "ReactComponent": [Function],
                    "__esModule": true,
                    "default": "GBP.svg",
                  }
                }
              />
              <p>
                GBP
              </p>
            </ForwardRef>
            <ForwardRef>
              Balance: 
              0
            </ForwardRef>
          </ForwardRef>
          <ForwardRef
            disabled={false}
            onChange={[Function]}
            placeholder="0"
            type="number"
          />
        </ForwardRef>,
        "style": Object {
          "background": "#FFFFFF",
          "color": "transparent",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              <ForwardRef>
                <ForwardRef
                  size="large"
                  src={
                    Object {
                      "ReactComponent": [Function],
                      "__esModule": true,
                      "default": "GBP.svg",
                    }
                  }
                />
                <p>
                  GBP
                </p>
              </ForwardRef>
              <ForwardRef>
                Balance: 
                0
              </ForwardRef>
            </ForwardRef>,
            <ForwardRef
              disabled={false}
              onChange={[Function]}
              placeholder="0"
              type="number"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef>
                  <ForwardRef
                    size="large"
                    src={
                      Object {
                        "ReactComponent": [Function],
                        "__esModule": true,
                        "default": "GBP.svg",
                      }
                    }
                  />
                  <p>
                    GBP
                  </p>
                </ForwardRef>,
                <ForwardRef>
                  Balance: 
                  0
                </ForwardRef>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <ForwardRef
                      size="large"
                      src={
                        Object {
                          "ReactComponent": [Function],
                          "__esModule": true,
                          "default": "GBP.svg",
                        }
                      }
                    />,
                    <p>
                      GBP
                    </p>,
                  ],
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "size": "large",
                      "src": Object {
                        "ReactComponent": [Function],
                        "__esModule": true,
                        "default": "GBP.svg",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-htpNat",
                        "isStatic": false,
                        "rules": Array [
                          "
	width: 100%;
	margin-right: 15px;
	height: ",
                          [Function],
                          "
",
                        ],
                      },
                      "displayName": "styled.img",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-htpNat",
                      "target": "img",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    },
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "GBP",
                    },
                    "ref": null,
                    "rendered": "GBP",
                    "type": "p",
                  },
                ],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-EHOje",
                    "isStatic": true,
                    "rules": Array [
                      "
	display: flex;
	align-items: center;
	height: 100%;
	background-color: transparent;
	cursor: pointer;
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-EHOje",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "Balance: ",
                    0,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Balance: ",
                  0,
                ],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bZQynM",
                    "isStatic": true,
                    "rules": Array [
                      "
	width: 20vw;
	color: black;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bZQynM",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": true,
                "rules": Array [
                  "
	display: flex;
	flex-direction: column;
	height: 100%;
	width: 30px;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": false,
              "onChange": [Function],
              "placeholder": "0",
              "type": "number",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
	width: 100%;
	font-size: 3rem; 
	appearance: none;
	margin: 0;
	background-color: transparent;
	text-align: right;
",
                ],
              },
              "displayName": "styled.input",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "input",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
	display: flex;
	align-items: center;
	height: 100%;
	padding: 0 4%;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
	display: flex;
	flex-direction: column;
	width: 100%;
	height: 10vh;
	background: ",
            [Function],
            ";
	padding: 5% 0;

	@media (min-width: 700px) {
		width: 45%;
	}
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
}
`;
